File: src/server/src/main/java/com/project/domain/user/Role.java
package com.project.domain.user;

import com.project.domain.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;

import java.util.UUID;

@Entity(name = "roles")
public class Role extends BaseEntity {
    @Id
    private UUID id;
    private String name;

    protected Role () {}

    public Role(String name) {
        super();
        this.id = UUID.randomUUID();
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public UUID getId() {
        return id;
    }
}

File: src/server/src/main/kotlin/com/project/domain/user/Role.kt
package com.project.domain.user

import com.project.domain.BaseEntity
import jakarta.persistence.Entity
import jakarta.persistence.Id
import java.util.UUID

@Entity(name = "roles")
class Role protected constructor(): BaseEntity() {
    @Id
    val id: UUID = UUID.randomUUID()
    var name: String? = null
        private set

    constructor(name: String) : this() {
        this.name = name
    }
}
